File: app/src/main/java/com/example/databindingdemo/view/MySeekBar.java
package com.example.databindingdemo.view;

import android.content.Context;
import android.databinding.InverseBindingListener;
import android.databinding.InverseBindingMethod;
import android.databinding.InverseBindingMethods;
import android.util.AttributeSet;

import com.xw.repo.BubbleSeekBar;

/**
 * @author yanglijun
 * @date 19-3-12
 */
@InverseBindingMethods({@InverseBindingMethod(type = MySeekBar.class, attribute = "app:progress",
        event = "progressAttrChanged")})
public class MySeekBar extends BubbleSeekBar {
    public MySeekBar(Context context) {
        super(context);
    }

    public MySeekBar(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public MySeekBar(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public void setProgressAttrChanged(final InverseBindingListener listener) {
        if (listener != null) {
            setOnProgressChangedListener(new OnProgressChangedListener() {
                @Override
                public void onProgressChanged(BubbleSeekBar bubbleSeekBar, int progress, float progressFloat, boolean fromUser) {
                    listener.onChange();
                }

                @Override
                public void getProgressOnActionUp(BubbleSeekBar bubbleSeekBar, int progress,
                                                  float progressFloat) {
                }

                @Override
                public void getProgressOnFinally(BubbleSeekBar bubbleSeekBar, int progress,
                                                 float progressFloat, boolean fromUser) {
                }
            });
        }
    }
}

File: app/src/main/java/com/example/databindingdemo/view/MySeekBar.kt
package com.example.databindingdemo.view

import android.content.Context
import android.databinding.InverseBindingListener
import android.databinding.InverseBindingMethod
import android.databinding.InverseBindingMethods
import android.util.AttributeSet

import com.xw.repo.BubbleSeekBar

/**
 * @author yanglijun
 * @date 19-3-12
 */
@InverseBindingMethods(InverseBindingMethod(type = MySeekBar::class, attribute = "progress", event = "progressAttrChanged"))
class MySeekBar : BubbleSeekBar {
    constructor(context: Context) : super(context) {}

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {}

    fun setProgressAttrChanged(listener: InverseBindingListener?) {
        if (listener != null) {
            onProgressChangedListener = object : BubbleSeekBar.OnProgressChangedListener {
                override fun onProgressChanged(bubbleSeekBar: BubbleSeekBar, progress: Int, progressFloat: Float, fromUser: Boolean) {
                    listener.onChange()
                }

                override fun getProgressOnActionUp(bubbleSeekBar: BubbleSeekBar, progress: Int,
                                                   progressFloat: Float) {
                }

                override fun getProgressOnFinally(bubbleSeekBar: BubbleSeekBar, progress: Int,
                                                  progressFloat: Float, fromUser: Boolean) {
                }
            }
        }
    }
}
